src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewTemplatingSystem.cs

```diff
@@ -0,0 +1,257 @@
+﻿
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Buffers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    internal class RazorViewTemplatingSystem : IViewTemplatingSystem
+    {
+        private readonly RazorViewLookup _viewLookup;
+        private readonly IRazorPageActivator _pageActivator;
+        
+        private readonly HtmlEncoder _htmlEncoder;
+        private readonly DiagnosticListener _diagnosticListener;
+        private readonly IReadOnlyList<IRazorPage> _viewStarts;
+        private IViewBufferScope _bufferScope;
+
+        public RazorViewTemplatingSystem(
+            RazorViewLookup viewLookup,
+            IRazorPageActivator razorPageActivator,
+            HtmlEncoder htmlEncoder,
+            DiagnosticListener diagnosticListener,
+            IRazorPage razorPage,
+            IReadOnlyList<IRazorPage> viewStarts)
+        {
+            _viewLookup = viewLookup;
+            _pageActivator = razorPageActivator;
+            _htmlEncoder = htmlEncoder;
+            _diagnosticListener = diagnosticListener;
+
+            Page = razorPage;
+            _viewStarts = viewStarts;
+        }
+
+        internal Action<IRazorPage, ViewContext> OnAfterPageActivated { get; set; }
+
+        private IRazorPage Page { get; }
+
+        private string Path => Page.Path;
+
+        public async Task InvokeAsync(ViewContext context)
+        {
+            _bufferScope = context.HttpContext.RequestServices.GetRequiredService<IViewBufferScope>();
```

Comment:
Most of the code that follows is copied from RazorView. We might be able to consolidate them, but that could be done later.